docs/content/users/extend/customizing-images.md

```diff
@@ -142,6 +151,33 @@ RUN chmod -R ugo+rw $COMPOSER_HOME
 ENV COMPOSER_HOME=""
 ```
 
+An example [Multi-stage](https://docs.docker.com/build/building/multi-stage/) web image could have a  `.ddev/web-build/stage.Dockerfile`:
+
+```dockerfile
+# We must declare the build enviroment variables we want to use on stage.Dockerfile
+ARG BASE_IMAGE="scratch"
+FROM $BASE_IMAGE AS build-stage-go
+
+# We must declare the build enviroment variables we want to use on stage.Dockerfile
+ARG uid
+ARG gid
+
+# install go
+RUN set -eux; \
+    GO_VERSION=1.24.1; \
+    AARCH=$(dpkg --print-architecture); \
+    wget -q https://go.dev/dl/go${GO_VERSION}.linux-${AARCH}.tar.gz -O go.tar.gz; \
```

Comment:
Automate to install the latest `go`:

```suggestion
    GO_VERSION=$(curl -fsSL "https://go.dev/dl/?mode=json" | jq -r ".[0].version"); \
    AARCH=$(dpkg --print-architecture); \
    wget -q https://go.dev/dl/${GO_VERSION}.linux-${AARCH}.tar.gz -O go.tar.gz; \
```

Reply:
@stasadev sure, I copied this over from my dockerfile, I used hard-coded versions so I can update them and make sure all my local layers are gonna be cached and to make it specific, but for the example, that's better.

Reply:
@stasadev I am doing something similar to get the latest version here https://github.com/hanoii/ddev-pimp-my-shell/blob/feb0b4f693573ec1338f1be0063c0b51f6625fd9/update.sh#L59